python/ray/train/torch/torch_predictor.py

```diff
@@ -44,17 +72,24 @@ def from_checkpoint(
             model: If the checkpoint contains a model state dict, and not
                 the model itself, then the state dict will be loaded to this
                 ``model``.
+            use_gpu: If set, the model will be moved to GPU on instantiation and
+                prediction happens on GPU.
         """
         model, preprocessor = load_checkpoint(checkpoint, model)
-        return TorchPredictor(model=model, preprocessor=preprocessor)
+        return TorchPredictor(model=model, preprocessor=preprocessor, use_gpu=use_gpu)
 
     def _predict_pandas(
         self,
         data: pd.DataFrame,
         dtype: Optional[Union[torch.dtype, Dict[str, torch.dtype]]] = None,
     ) -> pd.DataFrame:
         def tensorize(numpy_array, dtype):
-            torch_tensor = torch.from_numpy(numpy_array).to(dtype)
+            if self.use_gpu:
+                torch_tensor = torch.from_numpy(numpy_array).to(
+                    device="cuda", dtype=dtype
+                )
+            else:
+                torch_tensor = torch.from_numpy(numpy_array).to(dtype)
```

Comment:
Nit: I think you can keep the `torch.from_numpy(numpy_array).to(dtype)` the same for both and just add a `torch_tensor = torch_tensor.to(device="cuda")` for the gpu case.